src/EthereumTransaction.js

```diff
@@ -0,0 +1,254 @@
+/*-
+ * ‌
+ * Hedera JavaScript SDK
+ * ​
+ * Copyright (C) 2020 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import Hbar from "./Hbar.js";
+import FileId from "./file/FileId.js";
+import Transaction, {
+    TRANSACTION_REGISTRY,
+} from "./transaction/Transaction.js";
+
+/**
+ * @namespace proto
+ * @typedef {import("@hashgraph/proto").proto.ITransaction} HashgraphProto.proto.ITransaction
+ * @typedef {import("@hashgraph/proto").proto.ISignedTransaction} HashgraphProto.proto.ISignedTransaction
+ * @typedef {import("@hashgraph/proto").proto.TransactionBody} HashgraphProto.proto.TransactionBody
+ * @typedef {import("@hashgraph/proto").proto.ITransactionBody} HashgraphProto.proto.ITransactionBody
+ * @typedef {import("@hashgraph/proto").proto.ITransactionResponse} HashgraphProto.proto.ITransactionResponse
+ * @typedef {import("@hashgraph/proto").proto.IEthereumTransactionBody} HashgraphProto.proto.IEthereumTransactionBody
+ * @typedef {import("@hashgraph/proto").proto.IAccountID} HashgraphProto.proto.IAccountID
+ */
+
+/**
+ * @typedef {import("bignumber.js").default} BigNumber
+ * @typedef {import("./account/AccountId.js").default} AccountId
+ * @typedef {import("./channel/Channel.js").default} Channel
+ * @typedef {import("./client/Client.js").default<*, *>} Client
+ * @typedef {import("./Timestamp.js").default} Timestamp
+ * @typedef {import("./transaction/TransactionId.js").default} TransactionId
+ * @typedef {import("long").Long} Long
+ */
+
+/**
+ * Create a new Hedera™ crypto-currency account.
+ */
+export default class EthereumTransaction extends Transaction {
+    /**
+     * @param {object} [props]
+     * @param {Uint8Array | FileId} [props.ethereumData]
```

Comment:
I'm not sure these are mutually exclusive.
Is it not the case that the RLP encoded `ethereumData.callData` will be set to empty in services where CallData is specified.
In that case a user should be able to specify both.
Which means you should have `ethereumData` and `callData` params

Reply:
Would then also need to add a setter and getter for callData

Reply:
Huh, I think you're right. I believe I misread the protobuf documentation when we went with this API. Will fix.